src/random_graph.rs

```diff
@@ -493,3 +494,67 @@ pub fn random_geometric_graph(
     };
     Ok(graph)
 }
+
+/// Generate an undirected configuration model graph.
+///
+/// The configuration model is a random graph with a given degree sequence.
+///
+/// The algorithm is based on "M.E.J. Newman, "The structure and function of complex networks",
+///  SIAM REVIEW 45-2, pp 167-256, 2003.".
+///
+/// The graph construction process may insert parallel edges and self-loops.
+/// However, the probability of parallel edges and self-loops tends to
+/// converge towards zero as the number of nodes increases.
+///
+/// This algorithm has a time complexity of :math:`O(n + m)`
+///
+/// :param list degree_sequence: A list of positive integers specifying the degree sequence
+/// :param list weights: An optional list of node weights. If specified, the length of the list
+/// must be equal to the length of the degree sequence
+/// :param int seed: An optional seed to use for the random number generator
+///
+/// :return: A PyGraph object
+/// :rtype: PyGraph
+#[pyfunction]
+#[pyo3(text_signature = "(degree_sequence, /, weights=None, seed=None)")]
+pub fn undirected_configuration_model(
+    py: Python,
+    degree_sequence: Vec<usize>,
+    weights: Option<Vec<PyObject>>,
+    seed: Option<u64>,
+) -> PyResult<graph::PyGraph> {
+    let default_fn = || py.None();
+    let mut rng: Pcg64 = match seed {
+        Some(seed) => Pcg64::seed_from_u64(seed),
+        None => Pcg64::from_entropy(),
+    };
+    let degree_sequence = match core_generators::DegreeSequence::new(degree_sequence) {
```

Comment:
Do you need to check this degree sequence with [Erdős–Gallai](https://en.wikipedia.org/wiki/Erd%C5%91s%E2%80%93Gallai_theorem)? Or do you return a graph that might not be a simple graph?

Reply:
The graph may not be simple. The generator in NetworkX works in a similar fashion: https://networkx.org/documentation/stable/reference/generated/networkx.generators.degree_seq.configuration_model.html#configuration-model